### PR TITLE
issue/3204 Added check for _isTrackable and skip bug warn for missing components

### PR DESCRIPTION
**Pull request by oliverfoster**
fxies https://github.com/adaptlearning/adapt_framework/issues/3204

### Added
* Check for `_isTrackable: false`
* Warn for missing components rather than fail

### CODE REVIEW

File: js/serializers/ComponentSerializer.js [hunk 1]
```diff
@@ -26,7 +27,14 @@ define([
         const components = isContainer ?
           model.findDescendantModels('component') :
           [model];
+
+        let indexOffset = 0;
```
Comment (tomgreenfield) [inline js/serializers/ComponentSerializer.js:31]: Is the offset needed?

Comment (oliverfoster) [reply js/serializers/ComponentSerializer.js]: no, my mistake, thanks.

**Review by danielghost — APPROVED** [review]
Resolves the setAttemptObject error.

**Review by tomgreenfield — APPROVED** [review]
:eye:

**Review by cahirodoherty-learningpool — APPROVED** [review]
👀

**Pull request merged**